### DISCUSSION TITLE
Let PocketBase generate unique value when given value is empty

### DISCUSSION THREAD

**Discussion by LickABrick** [pocketbase/pocketbase#1752] [Q&A] [answered]
In my app I would like to make it possible to invite users to a group via a invite code (6 digit number in this case).
I don't want to make it possible for the user to make up this invite code themselves. 

I know it is possible to restrict a user from modifying certain fields by using the api rule: @request.data.*:isset = false
But ofcourse the field needs to have a value. Is it possible to let PocketBase generate a random value for me if I don't pass one myself? Or maybe like AppWrite does it, you can pass Unique() as a value and AppWrite will replace it with unique value.

**Comment by ganigeorgiev** [comment, marked as answer]
This is difficult to generalize (_unique/random of what characters set, length, prefix, suffix, etc._) and it is better left to the developers to handle programmatically on their own via the [Model](https://pocketbase.io/docs/event-hooks/#db-hooks) or [Record](https://pocketbase.io/docs/event-hooks/#record-crud-api-hooks) hooks. Something like:

```go
package main

import (
	"log"

	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/core"
	"github.com/pocketbase/pocketbase/models"
	"github.com/pocketbase/pocketbase/tools/security"
)

func main() {
	app := pocketbase.New()

	app.OnModelBeforeCreate("yourCollection").Add(func(e *core.ModelEvent) error {
		record := e.Model.(*models.Record)

		record.Set("code", security.RandomStringWithAlphabet(6, "0123456789"))

		return nil
	})

	if err := app.Start(); err != nil {
		log.Fatal(err)
	}
}

```


Eventually we may add a new read-only field type "Random" with various configuration options but I'm not really sure if it is necessary since you already have the option to generate the random value client-side or server-side via hook.

**Comment by LickABrick** [comment]
Thanks as always for your insanely fast and well explained responses!
